### PR TITLE
docs: Update docs to make isvc focal

### PR DESCRIPTION
**Pull request by pvaneck**
#### Motivation

The InferenceService CRD has evolved enough to become the primary interface for interacting with ModelMesh.
The documentation should reflect that.

#### Modifications

Documentation was adjusted to make the InferenceService CRD focal as opposed to the previous Predictor CRD.
Examples and snippets were added for InferenceServices.

#### Result

Users will learn and become more familiar with deploying on ModelMesh using the KServe InferenceService.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -11,9 +11,9 @@ The model data itself is pulled from one or more external [storage instances](pr
...
+- [`InferenceService`](predictors/) - This is the main interface KServe uses for managing models on Kubernetes. ModelMesh Serving can be used for deploying `InferenceService` predictors which represent a logical endpoint for serving predictions using a particular model. The `InferenceService` predictor spec specifies the model format, the storage location in which the model resides, and other optional configuration. The corresponding endpoint is "stable" and will seamlessly transition between different model versions or types when the spec is updated. Note that many features like transformers, explainers, and canary rollouts do not currently apply or fully work using InferenceServices with `deploymentMode` set to `ModelMesh`. And `PodSpec` fields that are set in the `InferenceService` predictor spec will be ignored.
```
Comment (chinhuang007) [inline docs/README.md:14]: I am not sure the "InferenceService link" should point to the ModelMesh predictors docs. I understand ModelMesh implements only the predictor component in the KServe InferenceService spec. But it feels a little strange to follow the link and see nothing about InferenceService. I would suggest either remove the link or have it point to KServe InferenceService.

File: docs/example-models.md [hunk 2]
```diff
@@ -2,9 +2,9 @@
 
 If ModelMesh Serving was deployed using `--quickstart`, a set of example models are shared via a MinIO instance to use when getting started with ModelMesh Serving and experimenting with the provided runtimes.
 
-## Predictors
+## InferenceServices
 
-The `config/example-predictors` directory contains Predictor manifests for many of the example models. Assuming that the entry specified below is added to the storage configuration secret, the Predictors can be deployed and used for experimentation.
+The `config/example-isvc` directory contains InferenceService manifests for many of the example models. Assuming that the entry specified below is added to the storage configuration secret, the InferenceServices can be deployed and used for experimentation.
```
Comment (chinhuang007) [inline docs/example-models.md:7]: Should it be `config/example-isvcs` directory?

File: docs/predictors/README.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
 # Using ModelMesh Serving
 
-Trained models are deployed in ModelMesh Serving via `Predictor`s. These represent a stable service endpoint behind which the underlying model can change.
+Trained models are deployed in ModelMesh Serving via `InferenceServices`s. The `predictor` component of `InferenceService` represent a stable service endpoint behind which the underlying model can change.
```
Comment (chinhuang007) [inline docs/predictors/README.md:3]: Nit: The `predictor` component of `InferenceService` represents....

File: docs/predictors/inferenceservice-cr.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+- `url` - URL holds the primary url that will distribute traffic over the provided traffic targets. This will be one the REST or gRPC endpoints that are available.
+- `restUrl` - REST endpoint of the component if available. This endpoint is provided through a REST proxy sidecar (if enabled), and this will also be the same for all predictors owned by a given ModelMesh Serving installation.
+- `grpcUrl` - gRPC endpoint of the component if available. Note that this will currently be the same for all InferenceService owned by a given ModelMesh Serving installation.
+
+`conditions` - Various condition entries. Pertinent entries are:
+
+- `PredictorReady`: predictor readiness condition. Status is `true` when the predictor's endpoints are ready to serve inferencing requests. Note that this does not _necessarily_ mean requests will respond immediately, the corresponding model may or may not be loaded in memory. In the case that it isn't there may be some delay before the response comes back.
+  - `Ready`: aggregated condition of all conditions.
+
```
Comment (chinhuang007) [inline docs/predictors/inferenceservice-cr.md:84]: Should "Ready" be at the level as "PredictorReady"?

File: docs/predictors/inferenceservice-cr.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    - `RuntimeNotRecognized` - There is no `ServingRuntime` defined with the specified runtime name.
+    - `InvalidPredictorSpec` - The current `InferenceService` predictor spec is invalid or unsupported.
+  - `location` - Indication of the pod in which a loading failure most recently occurred, if applicable. Its value will be the last 12 digits of the pod's full name.
+  - `message` - A message containing more detail about the error/failure.
+  - `modelId` - The internal id of the model in question. This includes a hash of the InferenceService's predictor spec.
+  - `time` - The time at which the failure occurred, if applicable.
+
+Upon creation, InferenceService the model status will always transition to `Loaded` state (unless the loading fails), but later if unused it is possible that they end up in a `Standby` state which means they are still available to serve requests but the first request could incur a loading delay. Whether this happens is a function of the available capacity and usage pattern of other models. It's possible that models will transition from `Standby` back to `Loaded` "by themselves" if more capacity becomes available.
```
Comment (chinhuang007) [inline docs/predictors/inferenceservice-cr.md:124]: Maybe just me, not quite sure
1. what "InferenceService the model status" means. I suppose it means the model status of the InferenceService?
2. what "they" refer to in "...they end up in a `Standby` state which means they are still available to serve requests..." I suppose they means InferenceServices?

File: docs/quickstart.md [hunk 6]
```diff
@@ -108,13 +134,45 @@ modelmesh-serving-mlserver-0.x-7db675f677-twrwd   3/3     Running   0          2
...
-example-mnist-predictor   sklearn   true        Loaded                      UpToDate     2m
+NAME                    URL                                               READY   PREV   LATEST   PREVROLLEDOUTREVISION   LATESTREADYREVISION   AGE
+example-sklearn-isvc    grpc://modelmesh-serving.modelmesh-serving:8033   True                                                                  97s
+```
+
+You can describe the `InferenceService` to get more status information:
+
+```shell
+kubectl describe example-sklearn-isvc
+
```
Comment (chinhuang007) [inline docs/quickstart.md:150]: Probably kubectl describe isvc example-sklearn-isvc?

File: docs/runtimes/custom_runtimes.md [hunk 7]
```diff
@@ -200,8 +200,8 @@ spec:
...
+In each entry of the `supportedModelFormats` list, `autoSelect: true` can optionally be specified to indicate that that the given `ServingRuntime` can be considered for automatic placement of `InferenceServices` with the corresponding model type/format if no runtime is explicitly specified.
+For example, if a user applies an `InferenceService` with `predictor.model.modelFormat.name: new-modeltype` and no `runtime` value, the above `ServingRuntime` will be used since it contains an "auto-selectable" supported model format that matches `new-modeltype`. If `autoSelect` were `false` or unspecified, the `InferenceService` would fail to load with the message "No ServingRuntime supports specified model type and/or protocol" unless the runtime `example-runtime` was specified directly in the YAML.
```
Comment (chinhuang007) [inline docs/runtimes/custom_runtimes.md:204]: Nit: maybe replace all instances of "new-modeltype" with "new-modelformat"?

**Review by chinhuang007 — COMMENTED** [review]
Thanks, @pvaneck ! Looks very good, just a few minor comments.

File: docs/predictors/inferenceservice-cr.md (hunk 5 above)
Comment (pvaneck) [reply docs/predictors/inferenceservice-cr.md:124]: Yea, it's not quite clear here. I will reword.

**Comment by pvaneck** [comment]
Thanks, @chinhuang007 for the thorough review!

**Review by chinhuang007 — APPROVED** [review]
/lgtm

**Pull request merged**